agroal-pool/src/main/java/io/agroal/pool/ConnectionHandler.java

```diff
@@ -185,12 +187,16 @@ public boolean isActive() {
         return stateUpdater.get( this ) == State.CHECKED_OUT;
     }
 
-    public long getLastAccess() {
-        return lastAccess;
+    public void touch() {
+        lastAccess = nanoTime();
+    }
+
+    public boolean isLeak(Duration timeout) {
+        return isActive() && !enlisted && isIdle( timeout );
```

Comment:
I'm assuming that if a connection is enlisted it's never to be considered a leak, as the transaction will eventually complete.

Reply:
Correct